controllers/component_build_controller.go

```diff
@@ -133,16 +133,6 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 
 	if err := r.SubmitNewBuild(ctx, component); err != nil {
-		// Try to revert the annotation
```

Comment:
I would not give up on any error. SubmitNewBuild could fail due to different reasons and not all of them lead to infinite loop.